src/engine/pieces/piece.ts

```diff
@@ -1,12 +1,30 @@
 import Player from '../player';
 import Board from '../board';
 import Square from '../square';
+import Direction from '../directions';
+
+export enum PieceType {
+    KING,
+    QUEEN,
+    ROOK,
+    BISHOP,
+    KNIGHT,
+    PAWN
+}
+
+export enum MoveDiagnostic {
```

Comment:
love the enum, still not 100% about the name, maybe something like `SquareAvailability` or `SquareState` as it is data about a specific square